src/react-imgix.jsx

```diff
@@ -220,14 +220,20 @@ class ReactImgix extends Component {
       defaultAttributeMap,
       this.props.attributeConfig
     );
-    const childProps = Object.assign({}, this.props.htmlAttributes, {
-      [attributeConfig.sizes]: this.props.sizes,
-      className: this.props.className,
-      width: width <= 1 ? null : width,
-      height: height <= 1 ? null : height,
-      [attributeConfig.src]: src,
-      ref: el => (this.imgRef = el)
-    });
+    const childProps = Object.assign(
+      {},
+      {
+        [attributeConfig.sizes]: this.props.sizes,
+        className: this.props.className,
+        width: width <= 1 ? null : width,
+        height: height <= 1 ? null : height,
+        [attributeConfig.src]: src,
+        ref: el => {
+          this.imgRef = el;
+        }
+      },
+      this.props.htmlAttributes
+    );
```

Comment:
I'm going to go out on a limb here and say that this is not the change we want. The problem here is that our own internal ref is not set when someone overrides the ref.

Instead, I think we should combine the ref handler into one handler. Something like this:

```jsx
ref: el => {
  this.imgRef = el; // handle internal ref
  
  setParentRef(this.props.htmlAttributes.ref); // handle ref from htmlAttributes
}

// elsewhere
const setParentRef = (ref) => {
  if (!ref) { return }

  // callback refs
  if (typeof ref === 'function') {
    ref(element);
  }

  // "modern" refs
  ref.current = element;
}
```

`setParentRef` needs to be a function that will handle `undefined`, and refs in the legacy and modern styles. You can find some more information on this, and various implementations, here: https://github.com/facebook/react/issues/13029. Be wary not to use hooks as that would violate our React version support range.

This package could also be useful: https://www.npmjs.com/package/@seznam/compose-react-refs

Reply:
Awesome thanks. If I understood your points correctly, I think it should now be able to account for each of these cases. Give it another look over and let me know.